### PR TITLE
feat(codepipeline-actions): Add detectChanges option to BitBucketSourceAction

### PR DESCRIPTION
**Pull request by bryan-hunter**
Adds optional `DetectChanges` to BitBucketSourceAction

https://docs.aws.amazon.com/codepipeline/latest/userguide/action-reference-CodestarConnectionSource.html

This option is available in the AWS console when configuring the Source stage.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by bryan-hunter** [comment]
@skinny85 no rush, but can I get a review when you get a chance :)

File: packages/@aws-cdk/aws-codepipeline-actions/test/bitbucket/bitbucket-source-action.test.ts [hunk 1]
```diff
@@ -80,11 +80,51 @@ nodeunitShim({
       },
     }));
 
+    test.done();
+  },
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/test/bitbucket/bitbucket-source-action.test.ts:84]: Add an empty line after this line.

File: packages/@aws-cdk/aws-codepipeline-actions/test/bitbucket/bitbucket-source-action.test.ts [hunk 2]
```diff
@@ -80,11 +80,51 @@ nodeunitShim({
...
+        },
+      ],
+    }));
+
     test.done();
   },
 });
 
-function createBitBucketAndCodeBuildPipeline(stack: Stack, props: { codeBuildCloneOutput: boolean }): void {
+function createBitBucketAndCodeBuildPipeline(stack: Stack, props: Partial<cpactions.BitBucketSourceActionProps>): void {
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/test/bitbucket/bitbucket-source-action.test.ts:127]: `Partial` eh? Fancy 🎩

File: packages/@aws-cdk/aws-codepipeline-actions/lib/bitbucket/source-action.ts [hunk 3]
```diff
@@ -63,6 +63,16 @@ export interface BitBucketSourceActionProps extends codepipeline.CommonAwsAction
...
+
+  /**
+   * Controls automatically starting your pipeline when a new commit
+   * is made on the configured repository and branch. If unspecified,
+   * the default value is true, and the field does not display by default.
+   *
+   * @default true
+   * @see https://docs.aws.amazon.com/codepipeline/latest/userguide/action-reference-CodestarConnectionSource.html
+   */
+  readonly detectChanges?: boolean;
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/lib/bitbucket/source-action.ts:75]: I'm not crazy about this name.

How about `triggerOnPush`?

**Review by skinny85 — DISMISSED** [review]
Looks great @bryan-hunter! I just have one question/comment about the name.

File: packages/@aws-cdk/aws-codepipeline-actions/lib/bitbucket/source-action.ts (hunk 3 above)
Comment (bryan-hunter) [reply packages/@aws-cdk/aws-codepipeline-actions/lib/bitbucket/source-action.ts]: changed to this

File: packages/@aws-cdk/aws-codepipeline-actions/test/bitbucket/bitbucket-source-action.test.ts (hunk 1 above)
Comment (bryan-hunter) [reply packages/@aws-cdk/aws-codepipeline-actions/test/bitbucket/bitbucket-source-action.test.ts:84]: done

**Review by skinny85 — APPROVED** [review]
Awesome @bryan-hunter , thanks so much for the contribution!

**Pull request merged**